docs/overview.rst

```diff
@@ -1,11 +1,410 @@
+      data.
+    * A :class:`Dataset` instance has the following useful attributes:
+
+      - ``sources``: tuple of strings indicating which sources are provided by
+        the dataset, and their ordering (which determines the return order of
+        :meth:`get_data`).
+      - ``axis_labels``: :class:`dict` mapping from source names to tuples of
+        strings or ``None``. Used to document the axis semantics of the dataset's
+        sources. Can be set via the ``axis_labels`` constructor argument.
+      - ``num_examples``: when available, represents the number of examples the
+        dataset holds.
+
+    * To use only a subset of a dataset's sources, set the ``sources``
+      constructor argument.
+    * You can interact with a dataset via the following methods:
+
+      - :meth:`open`: returns a ``state`` object the dataset will interact with
+        (e.g. a file handle), or ``None`` if it doesn't need to interact with
+        anything.
+      - :meth:`get_data`: given the ``state`` object and an optional ``request``
+        argument, returns data.
+      - :meth:`close`: given the ``state`` object, properly closes it.
+      - :meth:`reset`: given the ``state`` object, properly closes it and
+        returns a fresh one.
+
+    * Use :class:`IterableDataset` to interface with iterable objects in
+      general.
+
+      - :class:`IterableDataset` accepts an ``iterables`` argument, which is
+        expected to be a :class:`dict` mapping from source names to their
+        corresponding iterable objects.
+      - The state :meth:`IterableDataset.open` returns is an iterator object.
+      - Its :meth:`get_data` method doesn't accept requests.
+      - As such, it can only iterate exaplewise and sequentially.
+
+    * Use :class:`IndexableDataset` to interface with indexable objects in
+      general.
+
+      - :class:`IndexableDataset` accepts an ``indexables`` argument, which is
+        expected to be a :class:`dict` mapping from source names to their
+        corresponding indexable objects.
+      - The state :meth:`IndexableDataset.open` returns is ``None``.
+      - Its :meth:`get_data` method expects a list of indices as its
+        ``request`` keyword argument.
+      - As such, it allows random access.
+
+The :class:`Dataset` class is responsible for interfacing with the data and
+handling data access requests. Subclasses of :class:`Dataset` specialize in
+certain types of data.
+
+Datasets contain one or more **sources** of data, such as an array of images,
+a list of labels, a dictionary specifying an ontology, etc. Each source in a
+dataset is identified by a unique name.
+
+IterableDataset
+^^^^^^^^^^^^^^^
+
+The simplest :class:`Dataset` subclass is :class:`IterableDataset`, which
+interfaces with iterable objects.
+
+It is created by passing a :class:`dict` mapping source names to their
+associated data and, optionally, a :class:`dict` mapping source names to tuples
+of axis labels.
+
```

Comment:
I suggest to add the names of the dicts, i.e.: It is created by passing a :class:`dict` `iterables` mapping source names to their associated data and, optionally, a :class:`dict` `axis_labels` mapping source names to tuples of axis labels.